shared/src/androidMain/kotlin/dev/johnoreilly/confetti/work/SessionNotificationSender.kt

```diff
@@ -0,0 +1,158 @@
+package dev.johnoreilly.confetti.work
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.graphics.BitmapFactory
+import android.os.Build
+import android.util.Log
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import com.apollographql.apollo3.cache.normalized.FetchPolicy
+import dev.johnoreilly.confetti.ConfettiRepository
+import dev.johnoreilly.confetti.auth.Authentication
+import dev.johnoreilly.confetti.fragment.SessionDetails
+import dev.johnoreilly.confetti.shared.R
+import dev.johnoreilly.confetti.utils.DateService
+import kotlinx.datetime.LocalDateTime
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.toInstant
+import kotlinx.datetime.toLocalDateTime
+import kotlin.time.Duration.Companion.minutes
+
+class SessionNotificationSender(
+    private val context: Context,
+    private val repository: ConfettiRepository,
+    private val dateService: DateService,
+    private val notificationManager: NotificationManagerCompat,
+    private val authentication: Authentication,
+) {
+
+    suspend fun sendNotification() {
+        Log.d("marcello", "notify")
+
+        val user = authentication.currentUser.value
+
+        val conferences = repository.conferences(FETCH_POLICY)
+            .data
+            ?.conferences
+            .orEmpty()
+
+        val bookmarks = conferences.map { conference ->
+            repository.bookmarks(
```

Comment:
I do not know much about Confetti's GraphQL setup but I suspect the code here is sub-optimal and forces all conferences to be loaded. Maybe @yschimke or @martinbonnin can help me here?

Reply:
We are only using the current conference. Stale.